mongodb-advanced/components/people-service.js

```diff
@@ -1,36 +1,27 @@
 /*!
  * Copyright (c) 2016-2017 Digital Bazaar, Inc. All rights reserved.
  */
-export default function factory($http, brAlertService) {
+export default function factory($http) {
   const service = {};
 
-  service.savePerson = (person, callback) => {
+  service.savePerson = (person) => {
     return $http.post('/people/', person)
       .then(response => {
-        callback();
-      })
-      .catch(err => {
-        brAlertService.add('error', err);
+        return response.data;
       });
   };
 
   service.getPeople = callback => {
     return $http.get('/people')
-      .then(result => {
-        callback(result.data);
-      })
-      .catch(err => {
-        brAlertService.add('error', err);
+      .then(response => {
+        callback(response.data);
```

Comment:
callback needs to be eliminated.

Reply:
I now have....
in people-component.js
```
self.getPeople = () => exPeopleService.getPeople(people =>
    self.savedPeople = people)
    .catch(err => brAlertService.add('error', err));
```
And in people.service...
```
service.getPeople = () => {
    return $http.get('/people')
      .then(response => response.data);
  };
```
I get no error, but the list of people is not getting updated. How do we time it do the list updates.

Reply:
Joel, it should be:
```
self.getPeople = () => exPeopleService.getPeople().then(people =>
    self.savedPeople = people)
    .catch(err => brAlertService.add('error', err));
```
as written, you are passing your arrow function in as an argument to `getPeople`.